sdk/servicebus/service-bus/test/batchReceiver.spec.ts

```diff
@@ -1084,12 +1087,228 @@ describe("Batching - disconnects", function(): void {
+
+    messages.length.should.equal(1, "Unexpected number of messages received.");
+  });
+
+  it("returns messages if receive in progress (receiveAndDelete)", async function(): Promise<void> {
+    // Create the sender and receiver.
+    await beforeEachTest(TestClientType.UnpartitionedQueue, "receiveAndDelete");
+
+    // The first time `receiveMessages` is called the receiver link is created.
+    // The `receiver_drained` handler is only added after the link is created,
+    // which is a non-blocking task.
+    await receiverClient.receiveBatch(1, { maxWaitTimeInMs: 1000 });
+    const receiverContext = (receiverClient as ReceiverImpl<ReceivedMessage>)["_context"];
+
+    if (!receiverContext.batchingReceiver!.isOpen()) {
+      throw new Error(`Unable to initialize receiver link.`);
+    }
+
+    // Send a message so we have something to receive.
+    await senderClient.send(TestMessage.getSample());
+
+    // Simulate a disconnect after a message has been received.
+    receiverContext.batchingReceiver!["_receiver"]!.once("message", function() {
+      setTimeout(() => {
+        // Simulate a disconnect being called with a non-retryable error.
+        receiverContext.namespace.connection["_connection"].idle();
+      }, 0);
+    });
+
+    // Purposefully request more messages than what's available
+    // so that the receiver will have to drain.
+    const messages1 = await receiverClient.receiveBatch(10, { maxWaitTimeInMs: 10000 });
+
+    messages1.length.should.equal(1, "Unexpected number of messages received.");
+
+    // Make sure that a 2nd receiveMessages call still works
+    // by sending and receiving a single message again.
+    await senderClient.send(TestMessage.getSample());
+
+    // wait for the 2nd message to be received.
+    const messages2 = await receiverClient.receiveBatch(1, { maxWaitTimeInMs: 5000 });
+
+    messages2.length.should.equal(1, "Unexpected number of messages received.");
+  });
+
+  it("throws an error if receive is in progress (peekLock)", async function(): Promise<void> {
+    // Create the sender and receiver.
+    await beforeEachTest(TestClientType.UnpartitionedQueue);
+
+    // The first time `receiveMessages` is called the receiver link is created.
+    // The `receiver_drained` handler is only added after the link is created,
+    // which is a non-blocking task.
+    await receiverClient.receiveBatch(1, { maxWaitTimeInMs: 1000 });
+    const receiverContext = (receiverClient as ReceiverImpl<ReceivedMessageWithLock>)["_context"];
+
+    if (!receiverContext.batchingReceiver!.isOpen()) {
+      throw new Error(`Unable to initialize receiver link.`);
+    }
+
+    // Simulate a disconnect after a message has been received.
+    receiverContext.batchingReceiver!["_receiver"]!.once("message", function() {
+      setTimeout(() => {
+        // Simulate a disconnect being called with a non-retryable error.
+        receiverContext.namespace.connection["_connection"].idle();
+      }, 0);
+    });
```

Comment:
Now that you're not sending a message, you can't wait for a message to be received before calling idle (or else idle never gets called).

Instead I think you can just call idle in this setTimeout.

Reply:
Makes sense!